api/lang/customResponses.go

```diff
@@ -49,6 +55,43 @@ func specs() (interface{}, error) {
 	return specs, nil
 }
 
+func getImplFiles() (interface{}, error) {
+	if lRunner.runner == nil {
+		return nil, nil
+	}
+	implementationFileListResponse, err := getImplementationFileList()
+	if err != nil {
+		return nil, err
+	}
+	return implementationFileListResponse.ImplementationFilePaths, nil
+}
+
+func putStubImpl(req *jsonrpc2.Request) (interface{}, error) {
+	var stubImplParams StubImpl
+	if err := json.Unmarshal(*req.Params, &stubImplParams); err != nil {
+		logger.APILog.Debugf("failed to parse request %s", err.Error())
+		return nil, err
+	}
+	if lRunner.runner == nil {
+		return nil, nil
+	}
+	var stepVals []*gm.ProtoStepValue
+	for _, stepText := range stubImplParams.StepTexts {
+		stepValue, err := parser.ExtractStepValueAndParams(stepText, false)
+		if err != nil {
+			return nil, err
+		}
+		protoStepValue := gauge.ConvertToProtoStepValue(stepValue)
+
+		stepVals = append(stepVals, protoStepValue)
+	}
+	putStubImplementationResponse, err := putStubImplementation(stubImplParams.ImplementationFilePath, stepVals)
+	if err != nil {
+		return nil, err
+	}
+	return putStubImplementationResponse, nil
```

Comment:
A list of [workspaceEdits](https://microsoft.github.io/language-server-protocol/specification#workspaceedit) should be sent back to VSCode. In LSP, All changes that need to be done on the files should be done with a workspaceEdit.